sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +15,75 @@ namespace core
 namespace http
 {
 
-class http_request
+namespace http_method
+{
+enum HttpMethod
 {
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  std::string name;
+  std::string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(std::string name, std::string value)
```

Comment:
I think it's better to write the constructor like this

```C++
Header(std::string name, std::string value) : name(std::move(name)), value(std::move(value)) {}
```

It will be more efficient.

 @BillyONeal Since you are very professional about C++, is this the recommended way of writing constructor?